### ISSUE TITLE
How do I control output folders and file names?

### ISSUE THREAD

**Issue by aradalvand** [shaka-project/shaka-streamer#133]
I couldn't find any relevant options in https://shaka-project.github.io/shaka-streamer/configuration_fields.html.
With Shaka Packager, I used to use `segment_template`, `playlist_name`, etc.

**Comment by mariocynicys** [comment]
These are the options you can play with: https://github.com/shaka-project/shaka-streamer/blob/c6c5b924b617fea1e0c52268368a9ed236a50f4b/streamer/pipeline_configuration.py#L292-L299

But media file names are hard coded to a predefined template: https://github.com/shaka-project/shaka-streamer/blob/c6c5b924b617fea1e0c52268368a9ed236a50f4b/streamer/output_stream.py#L64-L89

**Comment by aradalvand** [comment]
Yes but none of those correspond to `segment_template` or `playlist_name` in Shaka Packager.
Setting `segment_folder` just moves all the segments into a single folder; whereas `segment_template` in Shaka Packager would've allowed you to put the segments of each stream into its own directory.
Likewise, `hls_output` corresponds to the `--hls_master_playlist_output` argument in Shaka Packager, which as the name implies determines the name of the master playlist. `playlist_name` in Shaka Packager on the other hand, specifies the name of the playlist file of an individual stream.

**Comment by joeyparrish** [comment]
It's not planned, but PRs are welcome.  We could maybe move the filename templates into the configuration, with the current templates as default.

**Comment by joeyparrish** [comment]
PRs still welcome!

**Comment by tomatotomata** [comment]
I was thinking of extending the existing `segment_folder` output option with configurable per-stream filename templates, while preserving the current defaults and leaving manifest naming unchanged. I would keep the change in `PipelineConfig` and `PackagerNode`, document the supported placeholders, and add focused config/argument tests. The goal is to cover the `segment_template` use case without changing the default output layout. Does that fit what you had in mind for this issue?
